tests/integration/targets/ec2_snapshot/tasks/main.yml

```diff
@@ -295,12 +292,36 @@
         that:
           - info_result.snapshots | length == 3
 
+    # delete the tagged snapshot - check mode
+    - name: Delete the tagged snapshot (check mode)
+      ec2_snapshot:
+        state: absent
+        snapshot_id: '{{ tagged_snapshot_id }}'
+      register: delete_result_check_mode
+      check_mode: true
+
+    - assert:
+        that:
+          - delete_result_check_mode is changed
+
     # delete the tagged snapshot
     - name: Delete the tagged snapshot
       ec2_snapshot:
         state: absent
         snapshot_id: '{{ tagged_snapshot_id }}'
 
+    # delete the tagged snapshot again (results in InvalidSnapshot.NotFound)
+    - name: Delete already removed snapshot (check mode)
+      ec2_snapshot:
+        state: absent
+        snapshot_id: '{{ tagged_snapshot_id }}'
+      register: delete_result_second_check_mode
+      check_mode: true
+
+    - assert:
+        that:
+          - delete_result_second_check_mode is not changed
+
```

Comment:
Minor niggle: please add a check for deleting an already removed snapshot not in checkmode.  If it errors just comment it out and open an issue.